dartagnan/src/test/java/com/dat3m/dartagnan/DataRaceTest.java

```diff
@@ -0,0 +1,131 @@
+        return Provider.fromSupplier(() -> new Settings(bound, 0));
+    }
+
+    protected Provider<Wmm> getWmmProvider() {
+        return GlobalSettings.ATOMIC_AS_LOCK ?
+                Providers.createWmmFromName(() -> "svcomp-locks") :
+                Providers.createWmmFromName(() -> "svcomp");
+    }
+
+    @ClassRule
+    public static CSVLogger.Initialization csvInit = CSVLogger.Initialization.create();
+
+    // Provider rules
+    protected final Provider<ShutdownManager> shutdownManagerProvider = Provider.fromSupplier(ShutdownManager::create);
+    protected final Provider<Arch> targetProvider = () -> Arch.NONE;
+    protected final Provider<String> filePathProvider = getProgramPathProvider();
+    protected final Provider<Settings> settingsProvider = getSettingsProvider();
+    protected final Provider<Program> programProvider = Providers.createProgramFromPath(filePathProvider);
+    protected final Provider<Wmm> wmmProvider = getWmmProvider();
+    protected final Provider<Result> expectedResultProvider = Provider.fromSupplier(() ->
+            readExpected(filePathProvider.get().substring(0, filePathProvider.get().lastIndexOf("-")) + ".yml"));
+    protected final Provider<VerificationTask> taskProvider = Providers.createTask(programProvider, wmmProvider, targetProvider, settingsProvider);
+    protected final Provider<SolverContext> contextProvider = Providers.createSolverContextFromManager(shutdownManagerProvider);
+    protected final Provider<ProverEnvironment> proverProvider = Providers.createProverWithFixedOptions(contextProvider, SolverContext.ProverOptions.GENERATE_MODELS);
+
+    // Special rules
+    protected final Timeout timeout = Timeout.millis(getTimeout());
+    protected final CSVLogger csvLogger = CSVLogger.create(() -> name);
+    protected final RequestShutdownOnError shutdownOnError = RequestShutdownOnError.create(shutdownManagerProvider);
+
+    @Rule
+    public RuleChain ruleChain = RuleChain.outerRule(shutdownManagerProvider)
+            .around(shutdownOnError)
+            .around(filePathProvider)
+            .around(settingsProvider)
+            .around(programProvider)
+            .around(wmmProvider)
+            .around(taskProvider)
+            .around(expectedResultProvider)
+            .around(csvLogger)
+            .around(timeout)
+            // Context/Prover need to be created inside test-thread spawned by <timeout>
+            .around(contextProvider)
+            .around(proverProvider);
+
+
+    private Result readExpected(String property) {
+        try (BufferedReader br = new BufferedReader(new FileReader(property))) {
+            while (!(br.readLine()).contains("no-data-race.prp")) {
+                continue;
+            }
+            return br.readLine().contains("false") ? FAIL : PASS;
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            System.exit(0);
+        }
+        return null;
+    }
```

Comment:
It might be better to move this method (and make it static) to `ResourceHelper`. At least I can imagine that all these `readExpected` methods should be reusable for various use-cases.